src/main/java/com/example/java/maven/ticketVendingMachine/CoinStorage.java

```diff
@@ -0,0 +1,86 @@
+package com.example.java.maven.ticketVendingMachine;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class CoinStorage {
+    private final Map<Coin, Integer> coins;
+    private final MessagePrinter messagePrinter;
+
+    public CoinStorage(Map<Coin, Integer> coins, MessagePrinter messagePrinter) {
+        this.coins = coins;
+        this.messagePrinter = messagePrinter;
+    }
+
+    public Map<Coin, Integer> getCoins() {
```

Comment:
when you need access `private final Map<Coin, Integer> coins` field inside `CoinStorage` you don't need this getter - you can simple write: `coins` or `this.coins` or even when you have method inside `CoinStorage`, for example:
```java
void doSth(CoinStorage otherCoinStorage){
   otherCoinStorage.coins
}
```

Reply:
Done